### PR TITLE
Feature: implement timeseries #4 #28

### LINKED ISSUE #4: Integrate MiD seasonal data
For more trip realism and variability it would be great to have the MiD seasonal probability distributions integrated into the tool.

- [x] add data to repo -> @Mosc5 you already have this data, right? Could you please add it?
- [ ] use it in simBEV, maybe using a simple param in the config to select the season as a first very simple solution

### LINKED ISSUE #28: simBEV only supports timerange of 14 days
A timerange of 14 days is currently hard-coded, the value from scenario cfg is not used.
This should be solved, maybe in combination with the seasonal data #4

### PR DESCRIPTION
**Pull request by Tk1999s**
included midtimeseries in simbev, so that simulations over more than 14 days are possible.

this PR closes #4 
and closes #28

### CODE REVIEW

**Comment by nesnoj** [comment]
Thanks for this super nice feature! I agreed with @khelfen that he'll do the review...

File: simbev/mid_timeseries.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+
+# set cutoff dates, jeweils erster Monat der nächsten Jahreszeit
+cutoff_dates = (3, 6, 9, 12)
+
+
+# returns season as string
+def get_season(date: datetime.date):
+    if date.month <= 2 or date.month == 12: season = "Winter"
+    elif date.month <= 5: season = "Fruehling"
```
Comment (khelfen) [inline simbev/mid_timeseries.py:15]: it's okay, but for me it's weird to have the `if` and the `TODO` in one line

File: simbev/mid_timeseries.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+import pandas as pd
+# import numpy as np
+import math
+import datetime
+from pathlib import Path
+
+
+# set cutoff dates, jeweils erster Monat der nächsten Jahreszeit
+cutoff_dates = (3, 6, 9, 12)
```
Comment (khelfen) [inline simbev/mid_timeseries.py:9]: better add hard coded variables into the config files and load them into the script

File: simbev/mid_timeseries.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            weeklist.append([get_season(start), math.floor(delta.days / 7), delta.days % 7, start, cutoff])
+        else:
+            delta = end - start + datetime.timedelta(1)
+            weeklist.append([get_season(start), math.floor(delta.days / 7), delta.days % 7, start,
+                             end + datetime.timedelta(1)])
+        start = cutoff
+
+    # set up variables
+    pd_result = pd.DataFrame()
+    weekday = 7
```
Comment (khelfen) [inline simbev/mid_timeseries.py:62]: better add hard coded variables into the config files

File: simbev/main_simbev.py [hunk 4]
```diff
@@ -79,6 +79,9 @@ def run_simbev(region_ctr, region_id, region_data, cfg_dict, charge_prob,
     ]
     count_cars = 0
 
+    region_path = main_path.joinpath(str(region_id))
+    region_path.mkdir(exist_ok=True)
```
Comment (khelfen) [inline simbev/main_simbev.py:83]: I prefare `os.makedirs()` over `path.mkdir()`. See [this](https://stackoverflow.com/a/13819575/13491957) answer on SO.

File: simbev/main_simbev.py [hunk 5]
```diff
@@ -146,73 +149,72 @@ def run_simbev(region_ctr, region_id, region_data, cfg_dict, charge_prob,
...
-                # print(demand.loc[demand.location == "6_home"].netto_charging_capacity)
-
-                # add results for this day to demand time series
-                # charging_all = charging_all.append(demand)
+            # for key in wd:
+            # create availability timeseries and charging times
+            (av, car_data, demand,
+             soc_start, idx_home,
+             idx_work, home_charging_capacity,
+             work_charging_capacity) = simbevMiD.availability(
```
Comment (khelfen) [inline simbev/main_simbev.py:157]: we really need to refactor this function :joy:

**Review by khelfen — CHANGES_REQUESTED** [review]
@Tk1999s Please add the `__pychache__` into the `.gitignore` file and remove those binary files from the PR. Also I added some small remarks but nothing big.

**Review by khelfen — APPROVED** [review]

**Comment by khelfen** [comment]
looks fine :) I can't merge it tough as I don't have the rights :sweat_smile:

**Pull request merged**